components/selector/__init__.py

```diff
@@ -35,7 +35,7 @@ class Selector(SchedulerComponent):
     def select(self,
                sites: FrozenSet[Site] = ALL_SITES,
                night_indices: Optional[npt.NDArray[NightIndex]] = None,
-               ranker: Optional[Ranker] = None) -> Selection:
+               ranker: Optional[DefaultRanker] = None) -> Selection:
```

Comment:
This looks good but I'm a bit confused by this line here. Wouldn't you want this to still be `Ranker` so you can pass in other subclasses of `Ranker`?

Reply:
My bad!